com.unity.netcode.gameobjects/Runtime/Connection/NetworkClient.cs

```diff
@@ -24,13 +59,28 @@ public List<NetworkObject> OwnedObjects
         {
             get
             {
-                if (PlayerObject != null && PlayerObject.NetworkManager != null && PlayerObject.NetworkManager.IsListening)
+                if (NetworkManager != null && NetworkManager.IsListening)
                 {
-                    return PlayerObject.NetworkManager.SpawnManager.GetClientOwnedObjects(ClientId);
+                    return NetworkManager.SpawnManager.GetClientOwnedObjects(ClientId);
                 }
 
                 return new List<NetworkObject>();
             }
         }
+        internal void AssignPlayerObject(ref NetworkObject networkObject)
+        {
+            PlayerObject = networkObject;
+        }
+
+        public NetworkClient(bool isServer, bool isClient, ulong clientId, NetworkManager networkManager)
+        {
+            SetRole(isServer, isClient, networkManager);
+            ClientId = clientId;
+        }
+
+        public NetworkClient()
```

Comment:
Is this actually valid? Creating a client that's neither server nor client?

Also, should these constructors be public? Do we actually expect users to create NetworkClients or should they be internal so we create them and then pass instances to users?

Reply:
The [NetworkClient ](https://github.com/Unity-Technologies/com.unity.netcode.gameobjects/blob/develop/com.unity.netcode.gameobjects/Runtime/Connection/NetworkClient.cs)has actually been public and part of NGO for awhile now.
The default role state of the NetworkClient (for this pass) is to be in a "neither a server-host or a client" until the NetworkManager is started and can set the context of the NetworkClient (i.e. StartServer, StartHost, StartClient).

Since the class has always been public, users could always construct a new instance (for what would be a good question)... so I guess a public constructor for a public class that has existed since MLAPI really has no impact on the API footprint (imo).

Reply:
Could we make the new constructor internal, and make the existing one throw an exception or at least log an error? The actual class being public makes perfect sense, it's just the constructors being public that feels off to me. Not a hill I'll die on though.

Reply:
we actually do not need to call `SetRole(false, false)` in the default public ctor because all properties are going to be initialized with their default values already (false, null, 0 etc.)

Reply:
in fact, I will delete both ctors and call `SetRole(...)` from the original call site.

Reply:
done, see c22dd73